tests/beignet/special/test__faddeeva_w.py

```diff
@@ -7,43 +9,28 @@
 
 @hypothesis.strategies.composite
 def _strategy(function):
-    x, y = torch.meshgrid(
-        torch.linspace(
-            function(
-                hypothesis.strategies.floats(
-                    min_value=-10,
-                    max_value=-10,
-                ),
-            ),
-            function(
-                hypothesis.strategies.floats(
-                    min_value=10,
-                    max_value=10,
-                ),
-            ),
-            steps=128,
-            dtype=torch.float64,
-        ),
-        torch.linspace(
-            function(
-                hypothesis.strategies.floats(
-                    min_value=-10,
-                    max_value=-10,
-                ),
-            ),
-            function(
-                hypothesis.strategies.floats(
-                    min_value=10,
-                    max_value=10,
-                ),
-            ),
-            steps=128,
-            dtype=torch.float64,
-        ),
-        indexing="xy",
+    dtype = function(hypothesis.strategies.sampled_from([torch.float64, torch.float32]))
+
+    if dtype == torch.float64:
+        rtol, atol = 1e-10, 1e-10
+    elif dtype == torch.float64:
+        rtol, atol = 1e-5, 1e-5
+
+    # avoid overflow of exp(y^2)
+    limit = math.sqrt(math.log(torch.finfo(dtype).max)) - 0.2
+
+    x = function(
+        hypothesis.strategies.floats(
+            min_value=-limit, max_value=limit, allow_nan=False, allow_infinity=False
```

Comment:
I understand disabling `infinity`, but why disable `nan`? Something funky with the semantics?

Reply:
hypothesis says you can't have `allow_nan=True` with `min_value` or `max_value` set

Reply:
I added a fix so we propagate nans through and a new test for this case.